pkg/controller/logstorage/logstorage.go

```diff
@@ -190,35 +173,21 @@ func (r *ReconcileLogStorage) createLogStorage(
 		KeyStoreSecret:              keyStoreSecret,
 	}
 
-	component := render.LogStorage(logStorageCfg)
+	logStorageComponent := render.LogStorage(logStorageCfg)
 
-	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, logStorageComponent); err != nil {
 		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
 		return reconcile.Result{}, false, finalizerCleanup, err
 	}
 
-	components = append(components, component,
-		rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
-			Namespace:       render.ElasticsearchNamespace,
-			ServiceAccounts: []string{render.ElasticsearchName},
-			KeyPairOptions: []rcertificatemanagement.KeyPairOption{
-				// We do not want to delete the secret from the tigera-elasticsearch namespace when CertificateManagement is
-				// enabled. Instead, it will be replaced with a TLS secret that serves merely to pass ECK's validation
-				// checks.
-				rcertificatemanagement.NewKeyPairOption(elasticKeyPair, true, elasticKeyPair != nil && !elasticKeyPair.UseCertificateManagement()),
-			},
-			TrustedBundle: trustedBundle,
-		}),
-	)
+	components = append(components, logStorageComponent)
 	if !operatorv1.IsFIPSModeEnabled(install.FIPSMode) {
+		// Render the key pair and trusted bundle into the Kibana namespace, which should be created by the log storage component above.
 		components = append(components,
 			rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
 				Namespace:       render.KibanaNamespace,
 				ServiceAccounts: []string{render.KibanaName},
 				KeyPairOptions: []rcertificatemanagement.KeyPairOption{
-					// We do not want to delete the secret from the tigera-elasticsearch when CertificateManagement is
```

Comment:
Why do you want to delete this comment? It is different from most other cases for a reason.

Reply:
Yeah, I deleted these I think because they didn't really make sense to me. I'll add it back.